engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodevalidation/NodeDataValidator.scala

```diff
@@ -0,0 +1,59 @@
+package pl.touk.nussknacker.engine.compile.nodevalidation
+
+import cats.data.ValidatedNel
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.MetaData
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.typed.typing.Typed
+import pl.touk.nussknacker.engine.compile.{ExpressionCompiler, NodeTypingInfo}
+import pl.touk.nussknacker.engine.graph.node.{Filter, NodeData}
+
+/*
+  Currently we only validate filter nodes. In the future we should implement validation/compilation for all node types
+  and refactor compiler accordingly, to avoid duplication
+ */
+trait NodeDataValidator[T<:NodeData] {
+
+  def compile(nodeData: T, validationContext: ValidationContext)(implicit metaData: MetaData): ValidationResponse
```

Comment:
Why `compile` instead of `validate`? It look like it doesn't produce any compiled output...